kolibri/core/webpack/hooks.py

```diff
@@ -0,0 +1,290 @@
+    # : For instance: "kolibri/core/assets/src/kolibri_core_app.js"
+    src_file = ""
+
+    # : The static directory where you want stuff to be written to
+    static_dir = "kolibri/core/static"
+
+    def __init__(self, *args, **kwargs):
+        super(WebpackBundleHook, self).__init__(*args, **kwargs)
+
+        # Verify the uniqueness of the slug
+        # It can be '0' in the parent class constructor
+        assert \
+            len([x.unique_slug == self.unique_slug for x in self.registered_hooks]) <= 1, \
+            "Non-unique slug found: '{}'".format(self.unique_slug)
+
+    class Meta:
+        abstract = True
+
+    @hooks.abstract_method
+    def get_by_slug(self, slug):
+        """
+        Fetch a registered hook instance by its unique slug
+        """
+        for hook in self.registered_hooks:
+            if hook.unique_slug == slug:
+                return hook
+        raise BundleNotFound("No bundle with that name is loaded: {}".format(slug))
+
+    @cached_property
+    @hooks.registered_method
+    def stats_file_content(self):
+        """
+        TODO: This property is only cached on the instance, maybe it should be
+        cached in a static module property instead so we can cache the JSON data
+        across the whole app?
+
+        :returns: A dict of the data contained in the JSON files which are
+        written by Webpack.
+        """
+        with open(self.stats_file) as f:
+            stats = json.load(f)
+        if django_settings.DEBUG:
+            timeout = 0
+            while stats['status'] == 'compiling':
+                time.sleep(getattr(settings, 'WEBPACK_POLL_INTERVAL', 0.1))
+                timeout += getattr(settings, 'WEBPACK_POLL_INTERVAL', 0.1)
+                with open(self.stats_file) as f:
+                    stats = json.load(f)
+                if timeout >= getattr(settings, 'WEBPACK_POLL_INTERVAL', 1.0):
+                    raise WebpackError('Webpack compilation still in progress')
+            if stats['status'] == 'error':
+                raise WebpackError('Webpack compilation has errored')
+        return {
+            "files": stats["chunks"][self.unique_slug]
+        }
+
+    @property
+    @hooks.registered_method
+    def bundle(self):
+        """
+        TODO: This is weird, why are we creating this as a dict object?
+
+        :returns: a generator yielding dict objects with properties of the built
+        asset, most notably its URL.
+        """
+        for f in self.stats_file_content["files"]:
+            filename = f['name']
+            if any(regex.match(filename) for regex in settings.IGNORE_PATTERNS):
+                continue
+            relpath = '{0}/{1}'.format(self.unique_slug, filename)
+            f['url'] = staticfiles_storage.url(relpath)
+            yield f
+
+    @hooks.registered_method
+    def bundle_filtered(self, extension=None):
+        """
+        TODO: Why is this helper function necessary!?
```

Comment:
In the unlikely, but not implausible event, that we want to explicitly only include JS files, or CSS files. If we don't have a strong use case, then we can delete.